### PR TITLE
[OpenMP] Rename worker threads for improved debuggability

### PR DESCRIPTION
**Pull request by HighW4y2H3ll**
Rename the worker threads "openmp_worker"

### CODE REVIEW

File: openmp/runtime/src/z_Linux_util.cpp [hunk 1]
```diff
@@ -878,6 +878,15 @@ void __kmp_create_worker(int gtid, kmp_info_t *th, size_t stack_size) {
     KMP_SYSFAIL("pthread_create", status);
   }
 
+#ifdef _GNU_SOURCE
```
Comment (WenleiHe) [inline openmp/runtime/src/z_Linux_util.cpp:881]: Is it a convention to check _GNU_SOURCE before __GLIBC__? Feels a bit unnecessary..

Also wondering if there is a more portable way of doing this.

File: openmp/runtime/src/z_Linux_util.cpp [hunk 2]
```diff
@@ -878,6 +878,15 @@ void __kmp_create_worker(int gtid, kmp_info_t *th, size_t stack_size) {
     KMP_SYSFAIL("pthread_create", status);
   }
 
+#ifdef KMP_OS_LINUX
+ #if ((__GLIBC__ > 2) || ((__GLIBC__ == 2) && (__GLIBC_MINOR__ >= 12)))
```
Comment (WenleiHe) [inline openmp/runtime/src/z_Linux_util.cpp:882]: nit: no indentation for macro

Comment (HighW4y2H3ll) [reply openmp/runtime/src/z_Linux_util.cpp]: updated! Thx :)

File: openmp/runtime/src/z_Linux_util.cpp [hunk 3]
```diff
@@ -878,6 +878,15 @@ void __kmp_create_worker(int gtid, kmp_info_t *th, size_t stack_size) {
...
   }
 
+#ifdef KMP_OS_LINUX
+#if ((__GLIBC__ > 2) || ((__GLIBC__ == 2) && (__GLIBC_MINOR__ >= 12)))
+  // Rename worker threads for improved debuggability
+  if (!KMP_UBER_GTID(gtid)) {
+    pthread_setname_np(handle, "openmp_worker");
+  }
+#endif
+#endif
```
Comment (MatzeB) [inline openmp/runtime/src/z_Linux_util.cpp:888]: You can simplify this
```suggestion
#if defined(KMP_OS_LINUX) && ((__GLIBC__ > 2) || ((__GLIBC__ == 2) && (__GLIBC_MINOR__ >= 12)))
  // Rename worker threads for improved debuggability
  if (!KMP_UBER_GTID(gtid)) {
    pthread_setname_np(handle, "openmp_worker");
  }
#endif
```

**Review by MatzeB — APPROVED** [review]
Added one nitpick. With that changed LGTM.

Please check the results of the CI and if they are good feel free to ping me to commit this.

**Review by WenleiHe — APPROVED** [review]
LGTM as well. May want to wait couple days before committing in case others have comments.

File: openmp/runtime/src/z_Linux_util.cpp [hunk 4]
```diff
@@ -878,6 +878,13 @@ void __kmp_create_worker(int gtid, kmp_info_t *th, size_t stack_size) {
     KMP_SYSFAIL("pthread_create", status);
   }
 
+#if defined(KMP_OS_LINUX) && ((__GLIBC__ > 2) || ((__GLIBC__ == 2) && (__GLIBC_MINOR__ >= 12)))
+  // Rename worker threads for improved debuggability
+  if (!KMP_UBER_GTID(gtid)) {
+    pthread_setname_np(handle, "openmp_worker");
```
Comment (shiltian) [inline openmp/runtime/src/z_Linux_util.cpp:884]: so other glibc version doesn't have this function or it doesn't work?

Comment (HighW4y2H3ll) [reply openmp/runtime/src/z_Linux_util.cpp:884]: `pthread_setname_np` is introduced in glibc 2.12, so yes, older version of glibc doesn't have this API.

Comment (shiltian) [reply openmp/runtime/src/z_Linux_util.cpp:884]: If this is the case, I'd suggest to use the CMake feature [`CheckSymbolExists`](https://cmake.org/cmake/help/latest/module/CheckSymbolExists.html#module:CheckSymbolExists).

Comment (HighW4y2H3ll) [reply openmp/runtime/src/z_Linux_util.cpp:884]: updated! :)

**Review by shiltian — APPROVED** [review]

**Pull request merged**